test/e2e/toolchainstatus_test.go

```diff
@@ -0,0 +1,94 @@
+package e2e
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	testconfig "github.com/codeready-toolchain/toolchain-common/pkg/test/config"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestForceMetricsSynchronization(t *testing.T) {
+	// given
+	toolchainClusterList := &toolchainv1alpha1.ToolchainClusterList{}
+	ctx, hostAwait, memberAwait, member2Await := WaitForDeployments(t, toolchainClusterList)
+	defer ctx.Cleanup()
+	hostAwait.UpdateToolchainConfig(
+		testconfig.AutomaticApproval().Enabled(),
+		testconfig.Metrics().ForceSynchronization(false))
+
+	userSignups := CreateMultipleSignups(t, ctx, hostAwait, memberAwait, 2)
+
+	// delete the current toolchainstatus/toolchain-status resource and restart the host-operator pod,
+	// so we can start with accurate counters/metrics (and not get flaky because of previous tests)
+	err := hostAwait.DeleteToolchainStatus("toolchain-status")
+	require.NoError(t, err)
+	err = hostAwait.DeletePods(client.InNamespace(hostAwait.Namespace), client.MatchingLabels{"name": "host-operator"})
+	require.NoError(t, err)
```

Comment:
May I know why? We shouldn't expect that the ToolchainStatus doesn't match the actual state/count of the resources. Otherwise, we wouldn't be able to verify the metrics in other tests, or am I missing something?

Reply:
that's right. I just wanted to make sure that we started the test with accurate metrics compared to existing resources, but indeed, they should already match. If it's a real concern, I can remove that part

Reply:
Understood - thanks for explanation :+1:

Reply:
btw, when the work on the `ToolchainConfig_controller` is finished, then we should drop the deletion of the pods. The `ToolchainConfig_controller` should watch the `ToolchainStatus` resource and recreate it. 
Could you please add a `TODO:` comment to this part?

@rajivnathan  do we have any Jira issue for this effort? - I mean watching the `ToochainStatus` CRD in `ToolchainConfig_controller`

Reply:
are we also going to trigger a full sync if the new `metrics.forceSynchronization` field is set to `true`?